static/scripts/rendering/write-add-remove.ts

```diff
@@ -0,0 +1,132 @@
+import { toastNotification } from "../../utils/toaster";
+import { ManifestRenderer } from "../render-manifest";
+import { Manifest, Plugin } from "../../types/plugins";
+import { parseConfigInputs } from "./input-parsing";
+import { getOfficialPluginConfig } from "../../utils/storage";
+
+export function writeNewConfig(renderer: ManifestRenderer, option: "add" | "remove"): void {
+  const selectedManifest = localStorage.getItem("selectedPluginManifest");
+  if (!selectedManifest) {
+    toastNotification("No selected plugin manifest found.", {
+      type: "error",
+      shouldAutoDismiss: true,
+    });
+    throw new Error("No selected plugin manifest found");
+  }
+  const pluginManifest = JSON.parse(selectedManifest) as Manifest;
+  const configInputs = document.querySelectorAll<HTMLInputElement | HTMLTextAreaElement>(".config-input");
+
+  const newConfig = parseConfigInputs(configInputs, pluginManifest);
+
+  renderer.configParser.loadConfig();
+
+  const officialPluginConfig: Record<string, { actionUrl?: string; workerUrl?: string }> = getOfficialPluginConfig();
+
+  const pluginName = pluginManifest.name;
+
+  // this relies on the manifest matching the repo name
+  const normalizedPluginName = pluginName
+    .toLowerCase()
+    .replace(/ /g, "-")
+    .replace(/[^a-z0-9-]/g, "")
+    .replace(/-+/g, "-");
+
+  const pluginUrl = Object.keys(officialPluginConfig).find((url) => {
+    return url.includes(normalizedPluginName);
+  });
+
+  if (!pluginUrl) {
+    toastNotification(`No plugin URL found for ${pluginName}.`, {
+      type: "error",
+      shouldAutoDismiss: true,
+    });
+    throw new Error("No plugin URL found");
+  }
+
+  const plugin: Plugin = {
+    uses: [
+      {
+        plugin: pluginUrl,
+        with: newConfig,
+      },
+    ],
+  };
+
+  if (option === "add") {
+    handleAddPlugin(renderer, plugin, pluginManifest);
+  } else {
```

Comment:
please use `else if (option === "remove")`, if new option is added someday can lead to error